diff --git a/tools/CurvePlot/src/cpp/CurvePlot.cxx b/tools/CurvePlot/src/cpp/CurvePlot.cxx
index d1a37150e..5334b9d25 100644
--- a/tools/CurvePlot/src/cpp/CurvePlot.cxx
+++ b/tools/CurvePlot/src/cpp/CurvePlot.cxx
@@ -159,7 +159,7 @@ namespace CURVEPLOT
        if(!PyUnicode_Check(ret_py))
          throw Exception("CurvePlot::toStdString(): Unexpected returned type!");
        Py_ssize_t size;
-       char *ptr = PyUnicode_AsUTF8AndSize(ret_py, &size);
+       char *ptr = (char *)PyUnicode_AsUTF8AndSize(ret_py, &size);
        ret_str = std::string(ptr);
     }
     return ret_str;
